### PR TITLE
[v1.8.x] Update LICENSE for all submodules.

### PR DESCRIPTION
**Pull request by josephevans**
## Description ##
This updates the LICENSE and NOTICE file to include the latest data from all submodules.

Part of #19427.

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by josephevans** [comment]
Ensured all items were correctly listed in our LICENSE file based on tvm's LICENSE file:

https://github.com/apache/incubator-tvm/blob/9bd2c7b44208ed992061f8c2688e1137357f1db1/LICENSE#L209-L240

File: NOTICE [hunk 1]
```diff
@@ -3,3 +3,14 @@
 
     This product includes software developed at
     The Apache Software Foundation (http://www.apache.org/).
+
+
+    --------------------------------------------------------------------------------
```
Comment (sandeep-krishnamurthy) [inline NOTICE:8]: Are there anyother submodules with Notice files?

**Review by sandeep-krishnamurthy — APPROVED** [review]
Thank you for this.

File: NOTICE (hunk 1 above)
Comment (josephevans) [reply NOTICE:8]: Nope

**Comment by sandeep-krishnamurthy** [comment]
@mxnet-bot run ci [windows-gpu]

**Review by mseth10 — APPROVED** [review]

File: LICENSE [hunk 2]
```diff
@@ -210,170 +210,176 @@
...
          Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
          Copyright (c) 2018 Open Neural Network Exchange
-    8. ONNX-TensorRT - For details, see 3rdparty/onnx-tensorrt/third_party/onnx/LICENSE
+    ONNX-TensorRT - For details, see 3rdparty/onnx-tensorrt/third_party/onnx/LICENSE
          Copyright (c) Facebook, Inc. and Microsoft Corporation.
-    9. clipboard.js - Refer to https://zenorocha.github.io/clipboard.js
+    clipboard.js - Refer to https://zenorocha.github.io/clipboard.js
          Licensed MIT © Zeno Rocha
-    10. clipboard.min.js - Refer to https://zenorocha.github.io/clipboard.js
+    clipboard.min.js - Refer to https://zenorocha.github.io/clipboard.js
```
Comment (leezu) [inline LICENSE:305]: > In LICENSE, add a pointer to the dependency's license **within the distribution** and a short note summarizing its licensing

https://infra.apache.org/licensing-howto.html

The pointer is not included in the distribution

Comment (samskalicky) [reply LICENSE:305]: what is needed? a path to the 3rdparty/... directory?

Comment (leezu) [reply LICENSE:305]: One needs to put the license somewhere in the distribution and point to it

Comment (leezu) [reply LICENSE:305]: The website can go down, that's why Apache wants it to be in the distribution

Comment (josephevans) [reply LICENSE:305]: It's the standard MIT license (included in Appendix), and the copyright line is included in LICENSE. Isn't this enough?

Comment (josephevans) [reply LICENSE]: Updated to reference source file, which contains copyright and license notice.

**Pull request merged**